Document wp_robots (#47)
humanmade/altis-seo: docs/robots.md

# Robots

## Robots API

> Altis v7 introduced a filter-based Robots API, providing central control over the `robots` meta tag. The `robots` meta tag allows you to utilize a more granular, page-specific approach to controlling how an individual page should be indexed and served to users in search engine results. The meta tag is usually placed in the `<head>` section of a page.

Reviewer: I don't think it's necessary to mention the version of altis here, we don't really do that anywhere in the docs other than the upgrade guide. Given we're moving to the docs site version with docs for all versions this shouldn't be an issue.
```suggestion
The Robots API provides central control over the `robots` meta tag. The `robots` meta tag allows you to utilize a more granular, page-specific approach to controlling how an individual page should be indexed and served to users in search engine results. The meta tag is automatically placed in the `<head>` section of a page.
```

```html
<!DOCTYPE html>
<html>
	<head>
		<meta name="robots" content="max-image-preview:large, follow" />
	</head>
```

> The Robots API allows you to hook into these meta tags to modify the tag's values. By default, the `robots` meta tag will include code that sets the maximum size of an image preview for images on the page. To disable this completely, use a `remove_filter` on the `wp_robots` filter hook.

Reviewer: It's always a single tag 
```suggestion
The Robots API allows you to hook into this meta tag to modify its values. By default, the `robots` meta tag will include code that sets the maximum size of an image preview for images on the page. To disable this completely, use a `remove_filter` on the `wp_robots` filter hook.
```

```php
remove_filter( 'wp_robots', 'wp_robots_max_image_preview_large' );
```

> You can interact and modify the contents of the `robots` meta tag using the `wp_robots` filter as well. The contents are passed into the filter as an array which can be interacted with.

Reviewer: ```suggestion
You can modify the contents of the `robots` meta tag using the `wp_robots` filter as well. The values are passed into the filter as an array.
```

```php
add_filter( 'wp_robots', function( array $robots ) : array {
	$robots['follow'] = true;
	$robots['foo'] = 'bar';
	unset( $robots['max-image-preview'] );

	return $robots;
} );
```

The example above would output the following:

```html
<meta name="robots" content="follow, foo:bar" />
```

Note that on local environments, and when the "Search engine visibility" setting in the admin Reading settings is set to "Discourage search engines from indexing this site", the `robots` meta tag will default to include `noindex, nofollow` unless overridden by the filter in addition to any custom parameters.

> For more information and examples, refer to the [`wp_robots` hook developer documentation](https://developer.wordpress.org/reference/hooks/wp_robots/).

Author: Should I link to the announcement post on the core P2 instead here, which has more (technical) info?
https://make.wordpress.org/core/2021/02/19/robots-api-and-max-image-preview-directive-in-wordpress-5-7/

Reviewer: I think this is ok to just reference the dev docs to be honest, we reference the field guide and release post in the upgrade guide.

Reviewer: We could perhaps provide a reference here for common robots meta tag values? Based off https://yoast.com/robots-meta-tags/ 

Not required as folks can look it up but make this a more useful reference given that the list of possible values isn't that long.

Author: Nice find. Added that in addition. I just wondered if linking to the announcement post -- which seems to have the greatest depth of technical information -- would be better than linking to the actual developer doc -- which hardly has anything, really. Really was just thinking about this after finding the announcement post for the lazy loading iframes feature which I don't think is really documented _anywhere_ else, since it doesn't have any real functions it adds.
